src/solve.jl

```diff
@@ -178,7 +178,7 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.
   # dtmin is all abs => does not care about sign already.
 
   if !isdae && isinplace(prob) && typeof(u) <: AbstractArray && eltype(u) <: Number && uBottomEltypeNoUnits == uBottomEltype && tType == tTypeNoUnits # Could this be more efficient for other arrays?
-    if !(typeof(u) <: ArrayPartition)
+    if typeof(u) <: ArrayPartition
       rate_prototype = recursivecopy(u)
     else
       rate_prototype = similar(u, typeof.(oneunit.(recursive_bottom_eltype.(u.x))./oneunit(tType))...)
```

Comment:
So maybe rather:
```suggestion
    rate_prototype = recursivecopy(u)
```
However, I didn't test it, so maybe it fixes the mentioned issue but breaks some other ArrayPartition cases...